classes/transaction.js

```diff
@@ -6,27 +6,32 @@ const {
   ecsign,
   toRpcSig,
   fromRpcSig,
+  pubToAddress,
+  bufferToHex,
+  privateToAddress
 } = require("ethereumjs-util");
 
 class Transaction {
-  constructor(to, signature, nonce, value, data, gasLimit, gasPrice) {
+  constructor(to, from, nonce, value, data, gasLimit, gasPrice) {
     this.to = to;
-    this.signature = signature;
+    this.from = from;
```

Comment:
Private key as the name implies always remains private with the entity who is holding the Public and Private key pair.
The only job of Private Key is to sign the transaction.
Signing does not imply that the signature will have the Private key in it. Digital signature where Private key acts like a random seed input to the Signing algorithm